pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularMedicalReleaseV2.java

```diff
@@ -0,0 +1,283 @@
+
+        List<FormSectionMembershipDto> v1SectionMembersAfterVersionUpdate = jdbiFormActivityFormSection
+                .findOrderedSectionMemberships(activityId, updatedV1VersionDto.getRevStart());
+        // Does section list look same before and after version update?
+        if (!extractSectionIds(v1SectionMembershipsBeforeVersionUpdate)
+                .equals(extractSectionIds(v1SectionMembersAfterVersionUpdate))) {
+            throw new DDPException("We messed up updating sections");
+        }
+
+
+        // Reading the new activity def. note that it is a complete activity! Not just a chunk!
+        FormActivityDef v2ActivityDefFromHocon = gson.fromJson(ConfigUtil.toJson(dataCfg), FormActivityDef.class);
+
+        // We are just interested in having a new revision of the contents; rest remains same
+        SectionBlockDao sectionBlockDao = handle.attach(SectionBlockDao.class);
+        sectionBlockDao.insertBodySections(activityId, v2ActivityDefFromHocon.getSections(), v2VersionDto.getRevId());
+
+        // checking we can read v2 back?
+        FormActivityDef activityDefV2 = findActivityDef(studyId, activityCode, "v2", handle);
+
+        // If we did it right, we should have a list of section ids that is different from v1
+        if (extractSectionIds(activityDefV2).equals(extractSectionIds(v1SectionMembersAfterVersionUpdate))) {
+            throw new DDPException("Sections do not appear to have been updated");
+        }
+
+        FormActivityDef activityDefV1 = findActivityDef(studyId, activityCode, "v1", handle);
+
+        // And V1 section ids should still look the same
+        if (!extractSectionIds(activityDefV1).equals(extractSectionIds(v1SectionMembersAfterVersionUpdate))) {
+            throw new DDPException("V1 sections are messed after adding v2 sections");
+        }
+
+        int originalV1QuestionCount = activityDefV1.getAllQuestions().size();
+        if (originalV1QuestionCount < 5) {
+            throw new DDPException("Oops. We should have a bunch of question in v1!");
+        }
+
+        long originalV1QuestionWithSuffix = activityDefV1.getAllQuestions().stream()
+                .filter(q -> q.getStableId().endsWith("_NEW")).count();
+        if (originalV1QuestionWithSuffix > 0) {
+            throw new DDPException("Oops! We had questions with suffix of _NEW already. This is not going to work");
+        }
+
+        // find the questions in v2 with _NEW suffix and replace them with corresponding one in v1
+        activityDefV2.getAllQuestions().stream()
+                .filter(q -> q.getStableId().endsWith("_NEW"))
+                .map(q -> new String[]{q.getStableId().substring(0, q.getStableId().length() - "_NEW".length()), q.getStableId()})
+                .forEach(toUseToDelete ->
+                        replaceQuestionInNewActivityVersion(activityCode, studyId, versionTag,
+                                "v1", toUseToDelete[1], toUseToDelete[0], handle));
+
```

Comment:
As the comment describes, the questions that I new I DID NOT want to keep from the new medical release v2 I gave them the suffix _NEW . Now I am going to find this suffix to find them, kill them, and replace them with the old questions from v1.